docs/compilation/performance_tips.md

```diff
@@ -0,0 +1,505 @@
+    fhe.BitwiseStrategy.CHUNKED,
+]
+
+for strategy in strategies:
+    compiler = fhe.Compiler(f, {"x": "encrypted", "y": "encrypted"})
+    circuit = compiler.compile(inputset, bitwise_strategy_preference=strategy)
+    print(
+        f"{strategy:>55} "
+        f"-> {circuit.programmable_bootstrap_count:>2} TLUs "
+        f"-> {int(circuit.complexity):>12_} complexity"
+    )
+```
+
+prints:
+
+```
+                       BitwiseStrategy.ONE_TLU_PROMOTED ->  1 TLUs ->  535_706_740 complexity
+                       BitwiseStrategy.THREE_TLU_CASTED ->  3 TLUs ->  599_489_229 complexity
+ BitwiseStrategy.TWO_TLU_BIGGER_PROMOTED_SMALLER_CASTED ->  2 TLUs ->  522_239_955 complexity
+ BitwiseStrategy.TWO_TLU_BIGGER_CASTED_SMALLER_PROMOTED ->  2 TLUs ->  519_246_216 complexity
+                                BitwiseStrategy.CHUNKED ->  6 TLUs ->  358_905_521 complexity
+```
+
+but if you change the inputset to:
+
+```python
+inputset = fhe.inputset(fhe.uint6, fhe.uint8)
+```
+
+it prints:
+
+```
+                       BitwiseStrategy.ONE_TLU_PROMOTED ->  1 TLUs ->  29_617_059_141 complexity
+                       BitwiseStrategy.THREE_TLU_CASTED ->  3 TLUs ->  89_750_593_728 complexity
+ BitwiseStrategy.TWO_TLU_BIGGER_PROMOTED_SMALLER_CASTED ->  2 TLUs ->  59_633_858_862 complexity
+ BitwiseStrategy.TWO_TLU_BIGGER_CASTED_SMALLER_PROMOTED ->  2 TLUs ->  59_633_858_862 complexity
+                                BitwiseStrategy.CHUNKED ->  6 TLUs ->   3_844_688_378 complexity
+```
+
+As you can see, strategies can affect the performance a lot! So make sure to select the appropriate one for your use case if you want to optimize performance.
+
+## Use round/truncate bit pattern before table lookups
+
+There are two extensions which can reduce the bit width of the table lookup input, [fhe.round_bit_pattern(...)](../core-features/rounding.md) and [fhe.truncate_bit_pattern(...)](../core-features/truncating.md), which can improve performance by sacrificing exactness.
+
+For example the following code:
+
+```python
+import numpy as np
+from concrete import fhe
+
+inputset = fhe.inputset(fhe.uint10)
+for lsbs_to_remove in range(0, 10):
+    def f(x):
+        return fhe.round_bit_pattern(x, lsbs_to_remove) // 2
+
+    compiler = fhe.Compiler(f, {"x": "encrypted"})
+    circuit = compiler.compile(inputset)
+
+    print(f"{lsbs_to_remove=} -> {int(circuit.complexity):>13_} complexity")
+```
+
+prints:
+
+```
+lsbs_to_remove=0 -> 9_134_406_574 complexity
+lsbs_to_remove=1 -> 3_209_430_092 complexity
+lsbs_to_remove=2 -> 1_536_476_735 complexity
+lsbs_to_remove=3 -> 1_588_749_586 complexity
+lsbs_to_remove=4 ->   848_133_081 complexity
+lsbs_to_remove=5 ->   525_987_801 complexity
+lsbs_to_remove=6 ->   358_276_023 complexity
+lsbs_to_remove=7 ->   373_311_341 complexity
+lsbs_to_remove=8 ->   400_596_351 complexity
+lsbs_to_remove=9 ->   438_681_996 complexity
+```
+
+## Activate approximate mode for rounding
+
```

Comment:
You should add a link to the approximate rounding in the doc. And maybe warns more that it less accurate

Reply:
I've added a link to the docs. It being less accurate is already written.